lib/spack/spack/installer.py

```diff
@@ -400,7 +400,11 @@ def dump_packages(spec, path):
 
             # There's no provenance installed for the source package.  Skip it.
             # User can always get something current from the builtin repo.
-            if not os.path.isdir(source_repo_root):
+            # If it's external, skip it because it either is using the builtin
```

Comment:
I would add "If" to the prior comment like:

> If there's no provenance installed for the source package...

since this is now an either/or situation.